modules/apps/batch-planner/batch-planner-rest-test/src/testIntegration/java/com/liferay/batch/planner/rest/resource/v1_0/test/FieldResourceTest.java

```diff
@@ -6,14 +6,267 @@
+				_toField(null, "permissions", false, "array", null),
+				_toField(null, "taxonomyCategoryIds", false, "array", "CSV"),
+				_toField(null, fieldName, false, "string", null),
+				_toField(
+					null, objectRelationship.getName(), false, "array", null)),
+			ListUtil.fromCollection(page.getItems()));
+	}
+
+	@Test
+	public void testGetFieldsWithMultipleOneToManyRelationship()
+		throws Exception {
+
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, true, null,
+						RandomTestUtil.randomString(),
+						"x" + RandomTestUtil.randomString(), false)),
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, true, null,
+						RandomTestUtil.randomString(),
+						"x" + RandomTestUtil.randomString(), false)),
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectRelationship objectRelationship1 =
+			ObjectRelationshipTestUtil.addObjectRelationship(
+				objectDefinition1, objectDefinition2,
+				TestPropsValues.getUserId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		ObjectDefinition objectDefinition3 =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				ObjectDefinitionTestUtil.getRandomName(),
+				Collections.singletonList(
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, true, null,
+						RandomTestUtil.randomString(),
+						"x" + RandomTestUtil.randomString(), false)),
+				ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		ObjectRelationship objectRelationship2 =
+			ObjectRelationshipTestUtil.addObjectRelationship(
+				objectDefinition3, objectDefinition2,
+				TestPropsValues.getUserId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		Page<Field> page = fieldResource.getPlanInternalClassNameKeyFieldsPage(
+			_getObjectDefinitionInternalClassName(objectDefinition2.getName()),
+			null);
+
+		String objectRelationship1Name = objectRelationship1.getName();
+
+		String objectRelationship1IdFieldName = StringBundler.concat(
+			"r_", objectRelationship1Name, "_",
+			objectDefinition1.getPKObjectFieldName());
+
+		String objectRelationship2Name = objectRelationship2.getName();
+
+		String objectRelationship2IdFieldName = StringBundler.concat(
+			"r_", objectRelationship2Name, "_",
+			objectDefinition3.getPKObjectFieldName());
+
+		int objectRelationshipFieldsCount = 0;
+
+		for (Field field : page.getItems()) {
```

Comment:
I updated the way we are validating by using the methods from the parent class. The loop is "never" a good idea when asserting the values